core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupTemplate.java

```diff
@@ -63,9 +66,31 @@ public void initializeLogging(LogConfig config) {
             handlers.add(handler);
             errorManager = handler.getLocalErrorManager();
         }
+
         if (config.file.enable) {
+            FileHandler handler = new FileHandler();
+            FileConfig.RotatingConfig rotationConfig = config.file.rotation;
+
+            if (rotationConfig.maxFileSize.isPresent() && rotationConfig.fileSuffix.isPresent()) {
+                PeriodicSizeRotatingFileHandler periodicSizeRotatingFileHandler = new PeriodicSizeRotatingFileHandler();
+                periodicSizeRotatingFileHandler.setSuffix(rotationConfig.fileSuffix.get());
+                periodicSizeRotatingFileHandler.setRotateSize(rotationConfig.maxFileSize.getAsLong());
+                periodicSizeRotatingFileHandler.setRotateOnBoot(rotationConfig.rotateOnBoot);
+                periodicSizeRotatingFileHandler.setMaxBackupIndex(rotationConfig.maxBackupIndex.getAsInt());
```

Comment:
If the max backup index is not present, then this will fail.

Reply:
If it is not actually optional then it should just be an `int`.  Otherwise the property should only be set if it is present.

Reply:
> If the max backup index is not present, then this will fail.

You are right. I thought the `defaultValue=1` I set will do the trick i.e my `OptionalInt` will always contain a value. I just run the same test without the property being set and the test fail. 


> If it is not actually optional then it should just be an `int`. Otherwise the property should only be set if it is present. 

Thanks. I'll go with an `int` having a defaultValue of `1`.